### ISSUE TITLE
Class extensions not working on modules

### ISSUE THREAD

**Issue by tmoran-stenoa** [ikhvorost/ReactBridge#2]
Hey!

Thank you so much for making this package. I've found a small issue: methods declared in extensions are not picked up by React. For example:

```swift
// TestModule.swift

import Foundation
import React
import ReactBridge

@ReactModule
class TestModule: NSObject, RCTBridgeModule {
  @ReactMethod
  @objc func sayHello() {
     print("Hello, world!")
  }
}

extension TestModule {
  @ReactMethod
  @objc func sayHello2() {
     print("Hello, world!")
  }
}
``` 

In the above, React is able to detect `sayHello`, but not `sayHello2`, so only the 1st can be invoked by the JS.

If I manually declare the methods, then it works:
```obj-c
#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(TestModule, NSObject)

RCT_EXTERN_METHOD(sayHello)
RCT_EXTERN_METHOD(sayHello2)

@end
```

Any idea what could be causing this?

**Comment by tmoran-stenoa** [comment]
Update: weirdly enough, if I explicitly add the static method which is generated by `@ReactMethod`, then it works:

```swift
extension TestModule {
 // React detects this!
  @objc static func __rct_export__sayHello2() -> UnsafePointer<RCTMethodInfo>? {
    struct Static {
      static let jsName = strdup("sayHello2")
      static let objcName = strdup("sayHello2")
      static var methodInfo = RCTMethodInfo(jsName: jsName, objcName: objcName, isSync: false)
    }
    return withUnsafePointer(to: &Static.methodInfo) {
        $0
    }
  }  

  @objc func sayHello2() {
     print("Hello, world!")
  }
}
```

**Comment by tmoran-stenoa** [comment]
> Hi! Looks like it is the compiler issue because it does't generate objc exported methods for React Native in the extensions. I saw the similar problems with Xcode 15.0 betas and hope it will be fixed in new releases of Xcode.

That explains a lot, thanks. Is there anywhere where we could report this bug?

**Comment by ikhvorost** [comment]
@tmoran-stenoa, the issue is fixed on Xcode 15.3 and you can declare your bridged methods within extensions.

**Comment by tmoran-stenoa** [comment]
> @tmoran-stenoa, the issue is fixed on Xcode 15.3 and you can declare your bridged methods within extensions.

Nice, thanks! I can't confirm this right now but will take your word for it.
